### PR TITLE
added suport for other income #seeSLPAI-66

### PR DESCRIPTION
**Pull request by mislavgrgic**
# Purpose

Adding other invoices to SDK

# Related Material

See Issue https://github.com/freshbooks/freshbooks-nodejs-sdk/issues/36

### CODE REVIEW

**Comment by mislavgrgic** [comment]
@amcintosh this is now ready for review, maybe some changes in the payload will be needed but we can put that to another ticket

**Review by amcintosh — APPROVED** [review]
Ya, we can do the payload separately. 🚢

File: packages/api/src/APIClient.ts [hunk 1]
```diff
@@ -284,6 +301,76 @@ export default class APIClient {
...
+				{
+					transformResponse: transformOtherIncomeResponse,
+				},
+				null,
+				'Get OtherIncome'
+			),
+		/**
+		 * Post other income
+		 */
+		create: (otherIncome: OtherIncome, accountId: string): Promise<Result<OtherIncome>> => {
```
Comment (seanmoon80) [inline packages/api/src/APIClient.ts:341]: Maybe `accountId` comes before `otherIncome` parameter like we discussed.

File: packages/api/src/APIClient.ts [hunk 2]
```diff
@@ -284,6 +301,76 @@ export default class APIClient {
...
+				`/accounting/account/${accountId}/other_incomes/other_incomes`,
+				{
+					transformResponse: transformOtherIncomeResponse,
+					transformRequest: transformOtherIncomeRequest,
+				},
+				otherIncome,
+				'Create OtherIncome'
+			)
+		},
+		update: (accountId: string, otherIncomeId: string, data: any): Promise<Result<OtherIncome>> =>
```
Comment (seanmoon80) [inline packages/api/src/APIClient.ts:353]: should be `OtherIncom` type instead of `data: any`?

File: packages/api/src/APIClient.ts (hunk 1 above)
Comment (mislavgrgic) [reply packages/api/src/APIClient.ts:341]: @seanmoon80 this is based on other `create` endpoints and all of them have it another way around so I think for the user, for now, is better if we keep it in that way

File: packages/api/src/APIClient.ts (hunk 2 above)
Comment (mislavgrgic) [reply packages/api/src/APIClient.ts:353]: for the update is not the whole object but just part of it. and based on the rest of the code I saw that it was always used and so I used that

**Pull request merged**